DEVELOPMENT.md

```diff
@@ -1,60 +1,172 @@
 
 ## Testing
 
-To test rustpython, there is a collection of python snippets located in the
-`tests/snippets` directory. To run those tests do the following:
+To test RustPython's functionality, a collection of Python snippets are located
+in the `tests/snippets` directory and can be run using `pytest`:
 
 ```shell
 $ cd tests
 $ pytest -v
 ```
 
-There also are some unit tests, you can run those with cargo:
+Rust unit tests can be run with `cargo`:
 
 ```shell
 $ cargo test --all
 ```
 
 ## Profiling
 
-To profile rustpython, simply build in release mode with the `flame-it` feature.
-This will generate a file `flamescope.json`, which you can then view at
+To profile RustPython, build it in `release` mode with the `flame-it` feature.
+This will generate a file `flamescope.json`, which can be viewed at
 https://speedscope.app.
 
-```sh
+```shell
 $ cargo run --release --features flame-it script.py
+```
+
+To display the default output json file:
+
+```shell
 $ cat flamescope.json
 {<json>}
 ```
 
-You can also pass the `--output-file` option to choose which file to output to
-(or stdout if you specify `-`), and the `--output-format` option to choose if
-you want to output in the speedscope json format (default), text, or a raw html
-viewer (currently broken).
+You can specify another file name other than the default by using the
+`--output-file` option to specify a file name (or `stdout` if you specify `-`).
+The `--output-format` option determines the format of the output file.
+The speedscope json format (default), text, or raw html can be passed. There
+exists a raw html viewer which is currently broken, and we welcome a PR to fix it.
+
+## Code organization
+
+Understanding a new codebase takes time. Here's a brief view of the
+repository's structure:
+
+- `bytecode/src`: python bytecode representation in rust structures
+- `compiler/src`: python compilation to bytecode
+- `derive/src`: Rust language extensions and macros specific to rustpython
+- `parser/src`: python lexing, parsing and ast
+- `Lib`: Carefully selected / copied files from CPython sourcecode. This is
+   the python side of the standard library.
+- `vm/src`: python virtual machine
+  - `builtins.rs`: Builtin functions
+  - `compile.rs`: the python compiler from ast to bytecode
+  - `obj`: python builtin types
+  - `stdlib`: Standard library parts implemented in rust.
+- `src`: using the other subcrates to bring rustpython to life.
+- `docs`: documentation (work in progress)
+- `py_code_object`: CPython bytecode to rustpython bytecode converter (work in
+  progress)
+- `wasm`: Binary crate and resources for WebAssembly build
+- `tests`: integration test snippets
+
+## Understanding Internals
+
+The RustPython package is the `rustpython` top-level crate. The `Cargo.toml`
+file in the root of the repo provide configuration of the crate and the
+implementation is found in the `src` directory (specifically, 
+`src/main.rs`).
```

Comment:
I'd say something about our cargo [workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) rather than a package, which I think is more Python terminology :)

Reply:
I've added mention of the workspace but I'm not sure that was where you wanted it mentioned.

Reply:
I just meant that paragraph, your change is good.